data_io.py

```diff
@@ -0,0 +1,70 @@
+import mimetypes
+import os
+from json import dump
+
+import pandas as pd
+
+
+# Input
+def take_input_data(path: str) -> pd.DataFrame:
+    if (
+        mimetypes.guess_type(path)[0] == "text/plain"
+    ):  # If it's a single file with text type use this block to simply load a df
+        df_data = pd.read_table(
+            path,
+            on_bad_lines="skip",
+            sep=" ",
+            skipinitialspace=True,
+            names=[
+                "Timestamp in seconds since the epoch",
+                "Response header size in bytes",
+                "Client IP address",
+                "HTTP response code",
+                "Response size in bytes",
+                "HTTP request method",
+                "URL",
+                "Username",
+                "Type of access/destination IP address",
+                "Response type",
+            ],
+        )
+    elif os.path.isdir(path):  # If it's a directory execute this block
```

Comment:
btw this function needs to be decoupled into load_data_from_directory and load_data_from_file

Reply:
That would make three of them though, won't it? 
- operator/manager: check input type, calls function responsible for file/folder, if neither return TypeError
- file func
- folder func

Would lengthen code but probably worth it. I will think about implementation and let you know.

Reply:
I'm staring at the code for like an hour and thinking of better way to write this function, and you're right, it validates single responsibility principle, is harder to test and maintain. On the other hand, it is probably most complicated part of the script and comments explain what's going on nicely. I personally like this function for that, even if it's not technically perfect. I'm open to changes if you can suggest detailed implementation; at worse it will stay, which I believe won't be that terrible?